partner_relations_in_tab/model/res_partner.py

```diff
@@ -0,0 +1,136 @@
+            rel_type.id,
+            'left' if not inverse else 'right')
+        field_names.append(field_name)
+        this_partner_name = '%s_partner_id' % (
+            'left' if not inverse else 'right')
+        other_partner_name = '%s_partner_id' % (
+            'left' if inverse else 'right')
+
+        field = etree.Element(
+            'field',
+            name=field_name,
+            context=('{"default_type_id": %s, "default_%s": id, '
+                     '"active_test": False}') % (
+                rel_type.id,
+                this_partner_name))
+        tab.append(field)
+        tree = etree.Element('tree', editable='bottom')
+        field.append(tree)
+        partner_type = rel_type.contact_type_left if inverse else \
+            rel_type.contact_type_right
+        if partner_type == 'c':
+            is_company = True
+        else:
+            is_company = False
+        tree.append(etree.Element(
+            'field',
+            string=_('Partner'),
+            domain=repr([('is_company', '=', is_company)]),
+            widget='many2one_clickable',
+            name=other_partner_name))
+        tree.append(etree.Element(
+            'field',
+            name='date_start'))
+        tree.append(etree.Element(
+            'field',
+            name='date_end'))
+        tree.append(etree.Element('field', name='type_id',
+                                  invisible='True'))
+        tree.append(etree.Element('field', name=this_partner_name,
+                                  invisible='True'))
+        return tab
+
+    def _add_relation_type_tab(
+            self, rel_type, inverse, field_names, relation_tab):
+        '''add the xml node to the view'''
+        tab = self._create_relation_type_tab(rel_type, inverse, field_names)
+        relation_tab.addnext(tab)
+
+    def fields_view_get(self, view_id=None, view_type='form', toolbar=False,
+                        submenu=False):
+        result = super(ResPartner, self).fields_view_get(
+            view_id=view_id, view_type=view_type,
+            toolbar=toolbar, submenu=submenu)
+        if view_type == 'form' and not self.env.context.get('check_view_ids'):
+            res_partner_relation_type = self.env['res.partner.relation.type']
+            own_tab_types = res_partner_relation_type.search(
+                ['|',
+                 ('own_tab_left', '=', True),
+                 ('own_tab_right', '=', True)
+                 ])
+            view = etree.fromstring(result['arch'])
+            relation_tab = view.xpath('//page[last()]')
+            if not relation_tab:
+                return result
+            relation_tab = relation_tab[0]
+            field_names = []
+            if not view.xpath('//field[@name="id"]'):
+                view.append(etree.Element('field', name='id',
+                                          invisible='True'))
+                field_names.append('id')
+            for rel_type in own_tab_types:
```

Comment:
Code can be mutch simplified if you just pass rel_type to underlying methods. No need then for separate function calls with inverse = True of inverse = False. inverse is translated back to left or right in underlying methods anyway, which just confuses things. (I know you just migrated the old code, and I think this was there as well, but now a good opportunity to clean up).